psl-core/src/main/java/org/linqs/psl/util/VizDataCollection.java

```diff
@@ -0,0 +1,233 @@
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.HashMap;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.HashSet;
+
+public class VizDataCollection {
+
+    private static VizDataCollection vizData = null;
+    private static Runtime runtime = null;
+
+    private JSONObject fullJSON;
+    private JSONArray predictionTruthArray;
+    private JSONArray ruleCountArray;
+    private JSONArray totRuleSatArray;
+    private JSONArray violatedGroundRulesArray;
+
+    JSONObject rules;
+    JSONObject groundRules;
+    JSONObject groundAtoms;
+
+    public static ArrayList<GroundRule> violatedGroundRulesList = new ArrayList<>();
+
+    static {
+        init();
+    }
+
+    // Static only.
+    private VizDataCollection() {
+        fullJSON = new JSONObject();
+        predictionTruthArray = new JSONArray();
+        ruleCountArray = new JSONArray();
+        totRuleSatArray = new JSONArray();
+        violatedGroundRulesArray = new JSONArray();
+        rules = new JSONObject();
+        groundRules = new JSONObject();
+        groundAtoms = new JSONObject();
+    }
+
+    private static synchronized void init() {
+        if (runtime != null) {
+            return;
+        }
+        vizData = new VizDataCollection();
+        runtime = Runtime.getRuntime();
+        runtime.addShutdownHook(new ShutdownHook());
+    }
+
+    //We want to make:
+    // A jsonObject filled with JSONArrays
+    // Will be organized into different JSON arrays that refer to specific modules, all in one object
+    //e.x.
+        // [
+        //     {predicate: Friends((bob,george), prediction: 0.00003, truth: 1}
+        //     {predicate: Friends((alice,george), prediction: 0.00003, truth: 1}
+        //     etc...
+        // ]
+    public static void outputJSON() {
+        vizData.fullJSON.put("PredictionTruth", vizData.predictionTruthArray);
+        vizData.fullJSON.put("RuleCount", vizData.ruleCountArray);
+        vizData.fullJSON.put("SatDis", vizData.totRuleSatArray);
+        vizData.fullJSON.put("ViolatedGroundRules", vizData.violatedGroundRulesArray);
+        vizData.fullJSON.put("rules", vizData.rules);
+        vizData.fullJSON.put("groundRules", vizData.groundRules);
+        vizData.fullJSON.put("groundAtoms", vizData.groundAtoms);
+
+        try (FileWriter file = new FileWriter("PSLVizData.json")) {
+            file.write(vizData.fullJSON.toString(4));
+            file.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static class ShutdownHook extends Thread {
+        @Override
+        public void run() {
+            outputJSON();
+        }
+    }
+
+    //Takes in a prediction truth pair and adds it to our map
+    public static void predictionTruth(GroundAtom target, float predictVal, float truthVal ) {
```

Comment:
I don't like this name.
It doesn't mean anything.

Change to something like "addTruth".

Reply:
updated.